File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/RemoteServiceConnector.java
package me.ycdev.android.lib.common.demo.service;

import android.content.Context;
import android.content.Intent;
import android.os.IBinder;
import androidx.annotation.NonNull;

import me.ycdev.android.lib.common.ipc.ServiceConnector;

public class RemoteServiceConnector extends ServiceConnector<IDemoService> {
    private static final String SERVICE_NAME = "RemoteService";

    public RemoteServiceConnector(Context cxt) {
        super(cxt, SERVICE_NAME);
    }

    @NonNull
    @Override
    protected Intent getServiceIntent() {
        return new Intent(mAppContext, RemoteService.class);
    }

    @Override
    protected IDemoService asInterface(IBinder service) {
        return IDemoService.Stub.asInterface(service);
    }
}

File: baseLib/src/androidTest/java/me/ycdev/android/lib/common/demo/service/RemoteServiceConnector.kt
package me.ycdev.android.lib.common.demo.service

import android.content.Context
import android.content.Intent
import android.os.IBinder
import androidx.annotation.NonNull

import me.ycdev.android.lib.common.ipc.ServiceConnector

open class RemoteServiceConnector(cxt: Context) :
    ServiceConnector<IDemoService>(cxt, SERVICE_NAME) {

    @NonNull
    public override fun getServiceIntent(): Intent {
        return Intent(mAppContext, RemoteService::class.java)
    }

    override fun asInterface(service: IBinder): IDemoService? {
        return IDemoService.Stub.asInterface(service)
    }

    companion object {
        private const val SERVICE_NAME = "RemoteService"
    }
}
